From: Danila Tikhonov <danila@jiaxyga.com>
Date: Tue, 24 May 2022 00:14:10 +0300
Subject: arm64: dts: qcom: sm7150: Add device-tree for Xiaomi Mi Note 10 Lite

Add device tree for the Xiaomi Mi Note 10 Lite (toco) smartphone.
This device is based on Snapdragon 730G (sm7150) SoC.

Tested-by: Igor Chernokalsky <chernokalsky89@gmail.com>
Signed-off-by: Danila Tikhonov <danila@jiaxyga.com>
---
 arch/arm64/boot/dts/qcom/Makefile                  |   1 +
 ...50-xiaomi-tucana.dts => sm7150-xiaomi-toco.dts} | 247 ++++-----------------
 2 files changed, 44 insertions(+), 204 deletions(-)
 copy arch/arm64/boot/dts/qcom/{sm7150-xiaomi-tucana.dts => sm7150-xiaomi-toco.dts} (57%)

diff --git a/arch/arm64/boot/dts/qcom/Makefile b/arch/arm64/boot/dts/qcom/Makefile
index dff425a..1dff8ba 100644
--- a/arch/arm64/boot/dts/qcom/Makefile
+++ b/arch/arm64/boot/dts/qcom/Makefile
@@ -277,6 +277,7 @@ dtb-$(CONFIG_ARCH_QCOM)	+= sm7150-xiaomi-davinci.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm7150-xiaomi-surya-huaxing.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm7150-xiaomi-surya-tianma.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm7150-xiaomi-sweet.dtb
+dtb-$(CONFIG_ARCH_QCOM)	+= sm7150-xiaomi-toco.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm7150-xiaomi-tucana.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm7225-fairphone-fp4.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= sm7325-nothing-spacewar.dtb
diff --git a/arch/arm64/boot/dts/qcom/sm7150-xiaomi-tucana.dts b/arch/arm64/boot/dts/qcom/sm7150-xiaomi-toco.dts
similarity index 57%
copy from arch/arm64/boot/dts/qcom/sm7150-xiaomi-tucana.dts
copy to arch/arm64/boot/dts/qcom/sm7150-xiaomi-toco.dts
index fd7f944..3aa09d7 100644
--- a/arch/arm64/boot/dts/qcom/sm7150-xiaomi-tucana.dts
+++ b/arch/arm64/boot/dts/qcom/sm7150-xiaomi-toco.dts
@@ -1,7 +1,6 @@
 // SPDX-License-Identifier: BSD-3-Clause
 /*
  * Copyright (c) 2024, Danila Tikhonov <danila@jiaxyga.com>
- * Copyright (c) 2024-2025, Alexander Baransky <alexander.baranskiy@yandex.ru>
  */
 
 /dts-v1/;
@@ -28,18 +27,14 @@ &tz_mem {
 };
 
 / {
-	/*
-	 * Xiaomi Mi Note 10 (Xiaomi Mi CC9 Pro)
-	 * Xiaomi Mi Note 10 Pro (Xiaomi Mi CC9 Pro Premium Edition)
-	 */
-	model = "Xiaomi Mi Note 10";
-	compatible = "xiaomi,tucana", "qcom,sm7150";
+	model = "Xiaomi Mi Note 10 Lite";
+	compatible = "xiaomi,toco", "qcom,sm7150";
 
-	qcom,board-id = <42 0>;
+	qcom,board-id = <44 0>;
 
 	chosen {
 		/*
-		 * Serial engine with the debug UART operates in SPI mode on tucana
+		 * Serial engine with the debug UART operates in SPI mode on toco
 		 * So disable console and remove stdout-path
 		 */
 		/delete-property/ stdout-path;
@@ -72,9 +67,9 @@ gpu_mem: gpu@97f15000 {
 			no-map;
 		};
 
-		rmtfs_mem: rmtfs@f2e01000 {
+		rmtfs_mem: rmtfs@f5e01000 {
 			compatible = "qcom,rmtfs-mem";
-			reg = <0 0xf2e01000 0 0x200000>;
+			reg = <0 0xf5e01000 0 0x200000>;
 			no-map;
 
 			qcom,client-id = <1>;
@@ -82,11 +77,35 @@ rmtfs_mem: rmtfs@f2e01000 {
 		};
 	};
 
+	vreg_ibb: regulator-ibb {
+		compatible = "regulator-fixed";
+		regulator-name = "ibb";
+		regulator-min-microvolt = <6000000>;
+		regulator-max-microvolt = <6000000>;
+	};
+
+	vreg_lab: regulator-lab {
+		compatible = "regulator-fixed";
+		regulator-name = "lab";
+		regulator-min-microvolt = <6000000>;
+		regulator-max-microvolt = <6000000>;
+	};
+
+	vreg_touch_vdd: regulator-touch-vdd {
+		compatible = "regulator-fixed";
+		regulator-name = "touch_vdd";
+		gpio = <&tlmm 72 GPIO_ACTIVE_HIGH>;
+		startup-delay-us = <4000>;
+		enable-active-high;
+		regulator-boot-on;
+		vin-supply = <&vreg_l10a_1p8>;
+	};
+
 	sound {
 		compatible = "qcom,sm8250-sndcard";
 		pinctrl-0 = <&pri_mi2s_active &pri_mi2s_ws_active>;
 		pinctrl-names = "default";
-		model = "Xiaomi Mi Note 10";
+		model = "Xiaomi Mi Note 10 Lite";
 
 		mm1-dai-link {
 			link-name = "MultiMedia1";
@@ -110,30 +129,6 @@ codec {
 			};
 		};
 	};
-
-	vreg_ibb: regulator-ibb {
-		compatible = "regulator-fixed";
-		regulator-name = "ibb";
-		regulator-min-microvolt = <6000000>;
-		regulator-max-microvolt = <6000000>;
-	};
-
-	vreg_lab: regulator-lab {
-		compatible = "regulator-fixed";
-		regulator-name = "lab";
-		regulator-min-microvolt = <6000000>;
-		regulator-max-microvolt = <6000000>;
-	};
-
-	vreg_touch_vdd: regulator-touch-vdd {
-		compatible = "regulator-fixed";
-		regulator-name = "touch_vdd";
-		gpio = <&tlmm 72 GPIO_ACTIVE_HIGH>;
-		startup-delay-us = <4000>;
-		enable-active-high;
-		regulator-boot-on;
-		vin-supply = <&vreg_l10a_1p8>;
-	};
 };
 
 &battery {
@@ -147,14 +142,14 @@ &framebuffer {
 
 &gpu {
 	zap-shader {
-		firmware-name = "qcom/sm7150/xiaomi/tucana/a615_zap.mbn";
+		firmware-name = "qcom/sm7150/xiaomi/toco/a615_zap.mbn";
 	};
 };
 
 &i2c4 {
 	clock-frequency = <100000>;
 	status = "okay";
-
+	
 	haptics@5a {
 		compatible = "ti,drv2605l";
 		reg = <0x5a>;
@@ -164,6 +159,8 @@ haptics@5a {
 		vib-rated-mv = <3200>;
 		vib-overdrive-mv = <3200>;
 	};
+
+	/* awinic,aw3644 (flash led) @ 63 */
 };
 
 &i2c7 {
@@ -195,6 +192,7 @@ touchscreen: fts@49 {
 		fts,x-max = <1080>;
 		fts,y-max = <2340>;
 	};
+
 	/* goodix,gt9886 (touchscreen) @ 5d */
 };
 
@@ -214,28 +212,12 @@ tfa9874: codec@34 {
 
 		#sound-dai-cells = <0>;
 	};
+
 	/* ti,bq2597x (charger) @ 66 */
 };
 
 &ipa {
-	firmware-name = "qcom/sm7150/xiaomi/tucana/ipa_fws.mbn";
-};
-
-&lpass_tlmm {
-	gpio-line-names = "SWR_TX_CLK", /* GPIO_0 */
-			  "SWR_TX_DATA1",
-			  "SWR_TX_DATA2",
-			  "SWR_RX_CLK",
-			  "SWR_RX_DATA1",
-			  "SWR_RX_DATA2",
-			  "WCD9375_RST_N",
-			  "NC",
-			  "NC",
-			  "NC",
-			  "NC", /* GPIO_10 */
-			  "NC",
-			  "BT_FM_SLIMBUS_DATA",
-			  "BT_FM_SLIMBUS_CLK";
+	firmware-name = "qcom/sm7150/xiaomi/toco/ipa_fws.mbn";
 };
 
 &nfc_default {
@@ -281,34 +263,6 @@ led-1 {
 	};
 };
 
-&pm6150_gpios {
-	gpio-line-names = "FOD_LDO_EN1", /* GPIO_1 */
-			  "FOD_LDO_EN2",
-			  "NC",
-			  "VBUS_NTC_CTRL",
-			  "SLB",
-			  "NC",
-			  "NC",
-			  "NC",
-			  "FORCED_USB_BOOT",
-			  "NC";  /* GPIO_10 */
-};
-
-&pm6150l_gpios {
-	gpio-line-names = "NC", /* GPIO_1 */
-			  "KYPD_VOLP_N",
-			  "CAMW_LDO_EN",
-			  "SF_DRV_STR",
-			  "NC",
-			  "NC",
-			  "SLB",
-			  "CAMT1_LDO_EN",
-			  "LCD_RESET_N",
-			  "PWM_HAPTICS", /* GPIO_10 */
-			  "CAMT2_LDO_EN",
-			  "SF_DRV_HWEN";
-};
-
 &q6afedai {
 	dai@16 {
 		reg = <PRIMARY_MI2S_RX>;
@@ -323,15 +277,15 @@ dai@0 {
 };
 
 &remoteproc_adsp {
-	firmware-name = "qcom/sm7150/xiaomi/tucana/adsp.mbn";
+	firmware-name = "qcom/sm7150/xiaomi/toco/adsp.mbn";
 };
 
 &remoteproc_cdsp {
-	firmware-name = "qcom/sm7150/xiaomi/tucana/cdsp.mbn";
+	firmware-name = "qcom/sm7150/xiaomi/toco/cdsp.mbn";
 };
 
 &remoteproc_mpss {
-	firmware-name = "qcom/sm7150/xiaomi/tucana/modem.mbn";
+	firmware-name = "qcom/sm7150/xiaomi/toco/modem.mbn";
 };
 
 &spi8 {
@@ -346,121 +300,6 @@ irled@0 {
 };
 
 &tlmm {
-	gpio-line-names = "NFC_ESE_SPI_MISO", /* GPIO_0 */
-			  "NFC_ESE_SPI_MOSI",
-			  "NFC_ESE_SPI_CLK",
-			  "NFC_ESE_SPI_CS_N",
-			  "HAPTICS_EN",
-			  "CAM_DOVDD_EN",
-			  "TS_I2C_SDA",
-			  "TS_I2C_SCL",
-			  "TS_RESET_N",
-			  "TS_INT_N",
-			  "MDP_VSYNC_P", /* GPIO_10 */
-			  "CAMT1_RSTN",
-			  "SPKR_PA_RST",
-			  "CAM_MCLK0",
-			  "CAM_MCLK1",
-			  "CAM_MCLK2",
-			  "CAM_MCLK3",
-			  "CCI_I2C_SDA0",
-			  "CCI_I2C_SCL0",
-			  "CCI_I2C_SDA1",
-			  "CCI_I2C_SCL1", /* GPIO_20 */
-			  "CAMM_LDO_EN",
-			  "FL_STROBE_TRIG",
-			  "CAMF_RSTN",
-			  "CAMM_RSTN",
-			  "CAMU_RSTN",
-			  "CAMW_RSTN",
-			  "CCI_I2C_SDA2",
-			  "CCI_I2C_SCL2",
-			  "CAMU_LDO_EN",
-			  "CAMT2_RSTN",  /* GPIO_30 */
-			  "NFC_LABBCLK3_EN",
-			  "ERR_INT_N",
-			  "LCD_ID_DET1",
-			  "NFC_I2C_SDA",
-			  "NFC_I2C_SCL",
-			  "NFC_DWL_REQ",
-			  "NFC_IRQ",
-			  "BT_UART_CTS",
-			  "BT_UART_RFR",
-			  "BT_UART_TX", /* GPIO_40 */
-			  "BT_UART_RX",
-			  "NC",
-			  "INFARED_SPI_MOSI",
-			  "BLSP2_UART_TX",
-			  "BLSP2_UART_RX",
-			  "APPS_I2C_SDA",
-			  "APPS_I2C_SCL",
-			  "FORCED_USB_BOOT",
-			  "SPKR_I2S_BCK",
-			  "SPKR_I2S_WS", /* GPIO_50 */
-			  "SPKR_I2S_DOUT",
-			  "SPKR_I2S_DIN1",
-			  "TYPC_I2C_SDA",
-			  "TYPC_I2C_SCL",
-			  "LCD_ID_DET2",
-			  "SPKR_INT",
-			  "CAMF_LDO_EN",
-			  "BQ25970_MASTER_INT",
-			  "FOD_SPI_MISO",
-			  "FOD_SPI_MOSI", /* GPIO_60 */
-			  "FOD_SPI_SCLK",
-			  "FOD_SPI_CS_N",
-			  "FOD_LDO_EN",
-			  "GPOUT",
-			  "FOD_RST",
-			  "SF_DRV_TX",
-			  "NFC_ENABLE",
-			  "TOF_INT",
-			  "FOD_INT",
-			  "RFFE6_CLK", /* GPIO_70 */
-			  "RFFE6_DATA",
-			  "TP_VDDI_EN",
-			  "WLAN_SW_CTRL",
-			  "GRFC_4",
-			  "UIM2_DATA",
-			  "UIM2_CLK",
-			  "UIM2_RESET",
-			  "UIM2_PRESENT",
-			  "UIM1_DATA",
-			  "UIM1_CLK", /* GPIO_80 */
-			  "UIM1_RESET",
-			  "UIM1_PRESENT",
-			  "GRFC_8",
-			  "GRFC_9",
-			  "WMSS_RESET_N",
-			  "ACCEL_INT",
-			  "GYRO_INT",
-			  "MIPI_SEL1",
-			  "ALS_INT_N",
-			  "MIPI_SEL2", /* GPIO_90 */
-			  "SF_DRV_TORCH",
-			  "RFFE4_CLK",
-			  "NC",
-			  "SW_IO",
-			  "GRFC_3",
-			  "QLINK_REQUEST",
-			  "QLINK_ENABLE",
-			  "GRFC_2",
-			  "NC",
-			  "NC", /* GPIO_100 */
-			  "RFFE3_DATA",
-			  "RFFE3_CLK",
-			  "RFFE4_DATA",
-			  "USB_PHY_PS",
-			  "RFFE5_DATA",
-			  "RFFE5_CLK",
-			  "GRFC_5",
-			  "WLAN_COEX_UART_TXD",
-			  "WLAN_COEX_UART_RXD",
-			  "GRFC_34", /* GPIO_110 */
-			  "GRFC_33",
-			  "RFFE1_DATA",
-			  "RFFE1_CLK";
-
 	spkr_reset: spkr-reset-state {
 		pins = "gpio12";
 		function = "gpio";
@@ -484,9 +323,9 @@ &uart8 {
 };
 
 &venus {
-	firmware-name = "qcom/sm7150/xiaomi/tucana/venus.mbn";
+	firmware-name = "qcom/sm7150/xiaomi/toco/venus.mbn";
 };
 
 &wifi {
-	qcom,ath10k-calibration-variant = "xiaomi_tucana";
+	qcom,ath10k-calibration-variant = "xiaomi_toco";
 };
